osxphotos/fileutil.py

```diff
@@ -2,21 +2,233 @@
+
+            # Skip dunder methods (e.g. __init__, __str__, etc.)
+            if name.startswith("__"):
+                continue
+
+            def make_retry(func):
+                @functools.wraps(func)
+                def wrapper(*args, **kwargs):
+                    retryer = retry(
+                        stop=stop_after_attempt(RETRY_FILEUTIL_CONFIG["retries"]),
+                        wait=wait_fixed(RETRY_FILEUTIL_CONFIG["wait_seconds"]),
+                        retry=retry_if_exception(is_fileutil_error),
+                        before_sleep=open_alias_script,
+                        # before=before_log(logger, logging.WARNING),
+                        after=after_log(logger, logging.WARNING),
+                        reraise=True,
+                    )
+                    return retryer(func)(*args, **kwargs)
+
+                return wrapper
+
+            # functions -> instance methods
+            if isinstance(value, types.FunctionType):
+                setattr(cls, name, make_retry(value))
+
+            # classmethod
+            elif isinstance(value, classmethod):
+                setattr(cls, name, classmethod(make_retry(value.__func__)))
+
+            # staticmethod
+            elif isinstance(value, staticmethod):
+                setattr(cls, name, staticmethod(make_retry(value.__func__)))
+
+        return cls
+
+    return decorator
+
+
+def open_alias_script(retry_state: RetryCallState) -> int | None:
+    """Attempt to (re)open a Finder alias pointing to an SMB export so macOS will
+    resolve and mount the associated network share.
+
+    Args:
+        retry_state: RetryCallState
+            Context/state object provided by the caller's retry mechanism. This function
+            accepts the value for compatibility with retry wrappers but does not use it
+            directly.
+
+    Returns:
+        int: The exit code returned by the 'osascript' subprocess when attempting
+            to run the AppleScript (0 indicates success, non-zero indicates failure).
+        None: Returned when no attempt is made because either the host is not
+            macOS or the configured alias path is empty.
+
+    Side effects:
+    - Reads the global RETRY_FILEUTIL_CONFIG["nas_export_alias"] to obtain the POSIX
+      path of the alias to open.
+    - Logs informational/warning messages about the attempt and outcome via the
+      global logger.
+    - If applicable, constructs and executes a small AppleScript that tells Finder
+      to open the POSIX file; Finder will resolve the alias and trigger mounting of
+      the SMB share if needed.
+
+    """
+
+    alias = RETRY_FILEUTIL_CONFIG.get("nas_export_alias", "")
+    logger.warning(
+        "⚠️  fileutil: SMB error: %s%s",
+        (
+            "retrying mount SMB alias"
+            if is_macos
+            else "not retrying mount SMB alias (not macOS)"
+        ),
+        (
+            f': --retry-nas-alias="{alias}"...'
```

Comment:
See previous comment about the export.py options in fileutil.py. Reference the cfg_fileutil_retry values, not the option names.